sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -3868,6 +3868,43 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
 
     assert(unions.size == 1)
   }
+
+  test("SPARK-34212 Parquet should read decimals correctly") {
+    // Decimal(2, 1) for INT32-backed decimal, Decimal(17, 2) for INT64-backed decimal
+    val df = sql("SELECT 1.0 a, CAST(100 AS DECIMAL(17, 2)) b, map(1, 2) c")
+
+    withTempPath { path =>
+      df.write.parquet(path.toString)
+
+      // 1. Complex schema disables vectorization automatically.
```

Comment:
Since we have case 2, why do we test case 1? To prove that complex schema disables vectorization? That's kind of something we already tested well and we don't need to repeat it here.

Reply:
Yep. I agree with you, @cloud-fan . Initially, this test case is designed to demonstrate that why some customers don't need to turn off the Spark's default configuration (enableVector...) in their environment. It's a leftover. Since the review is finished, I'll remove this ASAP.